source/Playnite/Database/GameDatabase.cs

```diff
@@ -1310,6 +1313,35 @@ bool updateCompletionStatus(Game game, CompletionStatusSettings settings)
             }
         }
 
+        public List<FilterPreset> GetSortedFilterPresets()
+        {
+            var filterPresetsSettings = (FilterPresets as FilterPresetsCollection).GetSettings();
+            var sortingOrder = filterPresetsSettings.SortingOrder;
+            var sortingDict = sortingOrder.ToDictionary(x => x, x => sortingOrder.IndexOf(x));
+
+            var savedSortedList = FilterPresets
+                .Where(x => sortingDict.ContainsKey(x.Id))
+                .OrderBy(x => sortingDict[x.Id])
+                .ToList();
+            var unsavedSortedList = FilterPresets
+                .Where(x => !sortingDict.ContainsKey(x.Id))
+                .OrderBy(x => x.Name)
+                .ToList();

```

Comment:
ToList is not necessary here since you are attaching this to another list anyways in the end.

Reply:
Done in https://github.com/JosefNemec/Playnite/pull/3258/commits/0c4a6c036edd6976cc0e4aad2bd0a27b423777bf